First angular assignment - chatting system

ksakthivel1211 opened the pull request:
All components and its functionalities where done according to the Ux given

ksakthivel1211 committed:
first-commit

ksakthivel1211 committed:
Message container was created

ksakthivel1211 committed:
Functionalities has been completed

All the functionalities for the Colour pallet , message bubble, and chat container components has been completed

ksakthivel1211 committed:
chat-bubble component

border radius style was added to chat bubble component

ksakthivel1211 committed:
outline changed for input

kavinprasath-cdw commented on assignment-1/src/app/colour-pallete-container/colour-pallete-container.component.html:
```diff
@@ -0,0 +1,9 @@
+<div class="colour-pallete-container">
+    <button class="colour-button clr1" value="#8f8f8f" (click)="onColorChange($event)" [ngClass]="{'selected':currentTheme==='#8f8f8f'}"></button>
```
we don't need a value attribute for this, this can be directly passed as input

ksakthivel1211 replied:
value was passed to the function

kavinprasath-cdw commented on assignment-1/src/app/chat-bubble/chat-bubble.component.css:
```diff
@@ -0,0 +1,30 @@
+    color: #c6c4c5;
+    text-align: end;
+    padding-top: 5px;
+    font-size: 10px;
```
nit: try to use variables for fonts and colours from SCSS

ksakthivel1211 replied:
Variables was used in css

kavinprasath-cdw commented on assignment-1/src/app/chat-bubble/chat-bubble.component.html:
```diff
@@ -0,0 +1,8 @@
+<div class="chat-bubble" [innerHTML]="singleMessage.message" 
+[ngClass]="{'right-chat-bubble':getBubbleSide()=='right','left-chat-bubble':getBubbleSide()=='left'}"
+[ngStyle]="{backgroundColor:getBubbleSide()=='right' ? getBubbleColor():'',borderRadius:getBubbleContainerSide()}"
```
nit: modify styles only with external styles, handle style changes with classes using ngClass and apply styles in the classes to improve code readability

ksakthivel1211 replied:
External styles where handled in css instead for writing the logic in type script.

kavinprasath-cdw commented on assignment-1/src/app/chat-bubble/chat-bubble.component.html:
```diff
@@ -0,0 +1,8 @@
+
+></div>
+<div class="time"
+[ngStyle]="{textAlign: getBubbleSide()=='right' ? 'right' : 'left'}"
```
This can be done in the style itself, i don't think we need build or listed logic for this

ksakthivel1211 replied:
External styles where handled in css instead for writing the logic in type script.

kavinprasath-cdw commented on assignment-1/src/app/message-container/message-container.component.html:
```diff
@@ -0,0 +1,30 @@
+  <div class="message-container">
+    <header>
+    <span class="back-icon-container"><img class="back-icon" src="../../assets/icons/chevron-left-solid.svg"></span> <img src="../../assets/images/wp6776560-minimalist-daredevil-wallpapers.jpg" class="chat-icon"> {{chatName}}
```
use constants for paths

ksakthivel1211 replied:
Image paths where moved to constants

kavinprasath-cdw commented on assignment-1/src/app/message-container/message-container.component.ts:
```diff
@@ -0,0 +1,85 @@
+
+  private receivingMessageValue!: message;
+
+  currentTime:any;
```
is this variable required?

ksakthivel1211 replied:
unused variable was removed

kavinprasath-cdw commented on assignment-1/src/app/message-container/message-container.component.ts:
```diff
@@ -0,0 +1,85 @@
+    return this.themeColour;
+  }
+
+  getCurrentTime():any{
```
i don't think we need to store this in a variable to get current time

ksakthivel1211 replied:
unused function was removed

kavinprasath-cdw reviewed (commented):
Try to add documented comments for functions and html

ksakthivel1211 committed:
review comenents

Attributes where passed as input value to function instead sending button value.

Variables where used for styles in css.
External styles where handled in css instead for writing the logic in type script.

All the ngStyle logics where removed.
Image paths where moved to constants.

ksakthivel1211 committed:
console logs removed

ksakthivel1211 committed:
Update .gitignore

ksakthivel1211 committed:
width adjustment

ksakthivel1211 committed:
removed console lod

ksakthivel1211 committed:
Merge branch 'first-assignment' of https://github.com/ksakthivel1211/Angular-training into first-assignment
